### DISCUSSION TITLE
CPPD model for Scene Text Recognition not supported | AssertionError: assert alg in [...]

### DISCUSSION THREAD

**Discussion by jaztiria** [PaddlePaddle/PaddleOCR#12988] [Q&A]
Hi all,

I'm trying to run the SOTA [CPPD model](https://github.com/PaddlePaddle/PaddleOCR/blob/dygraph/doc/doc_en/algorithm_rec_cppd_en.md) fro Scene Text Recognition, but I'm affraid it is not fully supported yet.

- System Environment：Ubuntu 22.04
- Related components：CPPD
- Command Code：`python3 -m paddle.distributed.launch --gpus '0' tools/eval.py -c ./rec_svtr_cppd_base_en_train/rec_svtrnet_cppd_base_en.yml -o Global.pretrained_model=./rec_svtr_cppd_base_en_train/best_model
`
- Complete Error Message：

> Traceback (most recent call last):
>   File "tools/eval.py", line 145, in <module>
>     config, device, logger, vdl_writer = program.preprocess()
>   File "xxx/scene-text-recognition-projects/PaddleOCR/tools/program.py", line 660, in preprocess
>     assert alg in [
> AssertionError

The problem is that at this point of the code the option `alg="CPPD"` is not supported.

```
assert alg in [
        'EAST', 'DB', 'SAST', 'Rosetta', 'CRNN', 'STARNet', 'RARE', 'SRN',
        'CLS', 'PGNet', 'Distillation', 'NRTR', 'TableAttn', 'SAR', 'PSE',
        'SEED', 'SDMGR', 'LayoutXLM', 'LayoutLM', 'LayoutLMv2', 'PREN', 'FCE',
        'SVTR', 'SVTR_LCNet', 'ViTSTR', 'ABINet', 'DB++', 'TableMaster', 'SPIN',
        'VisionLAN', 'Gestalt', 'SLANet', 'RobustScanner', 'CT', 'RFL', 'DRRG',
        'CAN', 'Telescope', 'SATRN', 'SVTR_HGNet'
    ]
```

Is there any workaround? If I try to add `"CPPD"` to the previous list, then I obtain the next error:

> File "xxx/scene-text-recognition-projects/PaddleOCR/ppocr/data/imaug/__init__.py", line 78, in create_operators
>     op = eval(op_name)(**param)
>   File "<string>", line 1, in <module>
> NameError: name 'CPPDLabelEncode' is not defined
> LAUNCH INFO 2023-11-29 10:57:54,271 Exit code 1

Thanks in advance,

Jon

**Comment by trantuankhoi** [comment]
https://github.com/PaddlePaddle/PaddleOCR/pull/10990

Have you update this code or pull from dygraph branch?
